PubSub/src/Subscription.php

```diff
@@ -722,10 +723,15 @@ public function acknowledgeBatch(array $messages, array $options = [])
     {
         $this->validateBatch($messages, Message::class);
 
-        $this->connection->acknowledge($options + [
-            'subscription' => $this->name,
-            'ackIds' => $this->getMessageAckIds($messages)
-        ]);
+        // the rpc may throw errors for a sub with EOD enabled
+        // but we don't act on the exception to maintain compatibility
+        try {
+            $this->connection->acknowledge($options + [
+                'subscription' => $this->name,
+                'ackIds' => $this->getMessageAckIds($messages)
+            ]);
+        } catch (BadRequestException $e) {
```

Comment:
Would we be able to only suppress this error if the reason is also `EXACTLY_ONCE_ACKID_FAILURE`?